### PR TITLE
drivers: adc: Place API into iterable section

### PR DESCRIPTION
**Pull request by pdgendt**
Move all adc driver api structs into an iterable section.

Depends on #71773

### CODE REVIEW

File: doc/releases/migration-guide-4.1.rst [hunk 1]
```diff
@@ -57,6 +57,13 @@ LVGL
 Device Drivers and Devicetree
 *****************************
 
+* Device driver APIs are placed into iterable sections (:github:`71773`) to allow for runtime
+  checking. See :ref:`device_driver_api` for more details.
+  The :c:macro:`DEVICE_API()` macro should be used by out-of-tree driver implementations for
+  the following driver classes:
+
+    * :c:struct:`adc_driver_api`
```
Comment (anangl) [inline doc/releases/migration-guide-4.1.rst:65]: Wouldn't it be better to do this in a separate PR for all modified driver classes after the corresponding PRs are merged?

**Review by anangl — APPROVED** [review]

**Review by kartben — APPROVED** [review]
